pkgs/development/python-modules/torchrl/default.nix

```diff
@@ -0,0 +1,157 @@
+{ lib
+, buildPythonPackage
+, pythonOlder
+, fetchFromGitHub
+, fetchpatch
+, ninja
+, setuptools
+, wheel
+, which
+, cloudpickle
+, numpy
+, torch
+, ale-py
+, gym
+, pygame
+, gymnasium
+, mujoco
+, moviepy
+, git
+, hydra-core
+, tensorboard
+, tqdm
+, wandb
+, packaging
+, tensordict
+, flaky
+, imageio
+, pytestCheckHook
+, pyyaml
+, scipy
+}:
+
+buildPythonPackage rec {
+  pname = "torchrl";
+  version = "0.2.1";
+  pyproject = true;
+
+  disabled = pythonOlder "3.8";
+
+  src = fetchFromGitHub {
+    owner = "pytorch";
+    repo = "rl";
+    rev = "refs/tags/v${version}";
+    hash = "sha256-Y3WbSMGXS6fb4RyXk2SAKHT6RencGTZXM3tc65AQx74=";
+  };
+
+  patches = [
+    (fetchpatch {  # https://github.com/pytorch/rl/pull/1828
+      name = "pyproject.toml-remove-unknown-properties";
+      url = "https://github.com/pytorch/rl/pull/1828/commits/3edbc3c8fb2452f370042131f57822291e80bd8a.patch";
+      hash = "sha256-4D5sN3HCD3zSAv2+9GddR9hORmQ1uzOaBZ4LoZwEBwM=";
+    })
+  ];
+
+  nativeBuildInputs = [
+    ninja
+    setuptools
+    wheel
+    which
+  ];
+
+  propagatedBuildInputs = [
+    cloudpickle
+    numpy
+    packaging
+    tensordict
+    torch
+  ];
+
+  passthru.optional-dependencies = {
+    atari = [
+      ale-py
+      gym
+      pygame
+    ];
+    gym-continuous = [
+      gymnasium
+      mujoco
+    ];
+    rendering = [
+      moviepy
+    ];
+    utils = [
+      git
+      hydra-core
+      tensorboard
+      tqdm
+      wandb
+    ];
+  };
+
+  # torchrl needs to create a folder to store datasets
+  preBuild = ''
+    export D4RL_DATASET_DIR=$(mktemp -d)
+  '';
+
+  pythonImportsCheck = [
+    "torchrl"
+  ];
+
+  # We have to delete the source because otherwise it is used instead of the installed package.
+  preCheck = ''
+    rm -rf torchrl
+
+    export XDG_RUNTIME_DIR=$(mktemp -d)
+  ''
+  # Otherwise, tochrl will try to use unpackaged torchsnapshot.
+  # TODO: This should be the default from next release so remove when updating from 0.2.1
+  + ''
+    export CKPT_BACKEND="torch"
+  '';
+
+  nativeCheckInputs = [
+    flaky
+    gymnasium
+    imageio
+    pytestCheckHook
+    pyyaml
+    scipy
+  ]
+  ++ passthru.optional-dependencies.atari
+  ++ passthru.optional-dependencies.gym-continuous
+  ++ passthru.optional-dependencies.rendering;
+
+  disabledTests = [
+    # TypeError: _FlakyPlugin._make_test_flaky() got an unexpected keyword argument 'reruns'
+    "test_parallel_env_transform_consistency"
+
+    # mujoco.FatalError: an OpenGL platform library has not been loaded into this process, this most likely means that a valid OpenGL context has not been created before mjr_makeContext was called
+    "test_vecenvs_env"
+
+    # ValueError: Can't write images with one color channel.
+    "test_log_video"
```

Comment:
@vmoens Do you have any idea why those tests are failing ?

Reply:
Can you point me to the error message?

Reply:
Sure:

<details>

```
==================================== ERRORS ====================================
_ ERROR at setup of TestParallel.test_parallel_env_transform_consistency[device0-4-Pendulum-v1] _
Traceback (most recent call last):
  File "/nix/store/xbsrnrzlv658h5r85pkw6v17ns63wlnx-python3.11-pytest-7.4.3/lib/python3.11/site-packages/_pytest/runner.py", line 341, in from_call
    result: Optional[TResult] = func()
                                ^^^^^^
  File "/nix/store/xbsrnrzlv658h5r85pkw6v17ns63wlnx-python3.11-pytest-7.4.3/lib/python3.11/site-packages/_pytest/runner.py", line 262, in <lambda>
    lambda: ihook(item=item, **kwds), when=when, reraise=reraise
            ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/zsnbysi65541mp54akjigfg9vw88xjvc-python3.11-pluggy-1.3.0/lib/python3.11/site-packages/pluggy/_hooks.py", line 493, in __call__
    return self._hookexec(self.name, self._hookimpls, kwargs, firstresult)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/zsnbysi65541mp54akjigfg9vw88xjvc-python3.11-pluggy-1.3.0/lib/python3.11/site-packages/pluggy/_manager.py", line 115, in _hookexec
    return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/zsnbysi65541mp54akjigfg9vw88xjvc-python3.11-pluggy-1.3.0/lib/python3.11/site-packages/pluggy/_callers.py", line 152, in _multicall
    return outcome.get_result()
           ^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/zsnbysi65541mp54akjigfg9vw88xjvc-python3.11-pluggy-1.3.0/lib/python3.11/site-packages/pluggy/_result.py", line 114, in get_result
    raise exc.with_traceback(exc.__traceback__)
  File "/nix/store/zsnbysi65541mp54akjigfg9vw88xjvc-python3.11-pluggy-1.3.0/lib/python3.11/site-packages/pluggy/_callers.py", line 77, in _multicall
    res = hook_impl.function(*args)
          ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/q5rcdhz2h20nxk38gsx8ydxjgwlvvp8j-python3.11-flaky-3.7.0/lib/python3.11/site-packages/flaky/flaky_pytest_plugin.py", line 244, in pytest_runtest_setup
    self._make_test_flaky(item, *marker.args, **marker.kwargs)
TypeError: _FlakyPlugin._make_test_flaky() got an unexpected keyword argument 'reruns'
=================================== FAILURES ===================================
___________________ TestGym.test_vecenvs_env[HalfCheetah-v4] ___________________
Traceback (most recent call last):
  File "/nix/store/xbsrnrzlv658h5r85pkw6v17ns63wlnx-python3.11-pytest-7.4.3/lib/python3.11/site-packages/_pytest/runner.py", line 341, in from_call
    result: Optional[TResult] = func()
                                ^^^^^^
  File "/nix/store/xbsrnrzlv658h5r85pkw6v17ns63wlnx-python3.11-pytest-7.4.3/lib/python3.11/site-packages/_pytest/runner.py", line 262, in <lambda>
    lambda: ihook(item=item, **kwds), when=when, reraise=reraise
            ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/zsnbysi65541mp54akjigfg9vw88xjvc-python3.11-pluggy-1.3.0/lib/python3.11/site-packages/pluggy/_hooks.py", line 493, in __call__
    return self._hookexec(self.name, self._hookimpls, kwargs, firstresult)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/zsnbysi65541mp54akjigfg9vw88xjvc-python3.11-pluggy-1.3.0/lib/python3.11/site-packages/pluggy/_manager.py", line 115, in _hookexec
    return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/zsnbysi65541mp54akjigfg9vw88xjvc-python3.11-pluggy-1.3.0/lib/python3.11/site-packages/pluggy/_callers.py", line 152, in _multicall
    return outcome.get_result()
           ^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/zsnbysi65541mp54akjigfg9vw88xjvc-python3.11-pluggy-1.3.0/lib/python3.11/site-packages/pluggy/_result.py", line 114, in get_result
    raise exc.with_traceback(exc.__traceback__)
  File "/nix/store/zsnbysi65541mp54akjigfg9vw88xjvc-python3.11-pluggy-1.3.0/lib/python3.11/site-packages/pluggy/_callers.py", line 77, in _multicall
    res = hook_impl.function(*args)
          ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/xbsrnrzlv658h5r85pkw6v17ns63wlnx-python3.11-pytest-7.4.3/lib/python3.11/site-packages/_pytest/runner.py", line 177, in pytest_runtest_call
    raise e
  File "/nix/store/xbsrnrzlv658h5r85pkw6v17ns63wlnx-python3.11-pytest-7.4.3/lib/python3.11/site-packages/_pytest/runner.py", line 169, in pytest_runtest_call
    item.runtest()
  File "/nix/store/xbsrnrzlv658h5r85pkw6v17ns63wlnx-python3.11-pytest-7.4.3/lib/python3.11/site-packages/_pytest/python.py", line 1792, in runtest
    self.ihook.pytest_pyfunc_call(pyfuncitem=self)
  File "/nix/store/zsnbysi65541mp54akjigfg9vw88xjvc-python3.11-pluggy-1.3.0/lib/python3.11/site-packages/pluggy/_hooks.py", line 493, in __call__
    return self._hookexec(self.name, self._hookimpls, kwargs, firstresult)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/zsnbysi65541mp54akjigfg9vw88xjvc-python3.11-pluggy-1.3.0/lib/python3.11/site-packages/pluggy/_manager.py", line 115, in _hookexec
    return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/zsnbysi65541mp54akjigfg9vw88xjvc-python3.11-pluggy-1.3.0/lib/python3.11/site-packages/pluggy/_callers.py", line 113, in _multicall
    raise exception.with_traceback(exception.__traceback__)
  File "/nix/store/zsnbysi65541mp54akjigfg9vw88xjvc-python3.11-pluggy-1.3.0/lib/python3.11/site-packages/pluggy/_callers.py", line 77, in _multicall
    res = hook_impl.function(*args)
          ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/xbsrnrzlv658h5r85pkw6v17ns63wlnx-python3.11-pytest-7.4.3/lib/python3.11/site-packages/_pytest/python.py", line 194, in pytest_pyfunc_call
    result = testfunction(**testargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/z3h53fn0184inh06pynjilh4q2psklx3-python3.11-torchrl-0.2.1/lib/python3.11/site-packages/torchrl/_utils.py", line 349, in _lazy_call_fn
    return self._delazify(self.func_name)(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/build/source/test/test_libs.py", line 465, in test_vecenvs_env
    env = GymEnv(envname, num_envs=2, from_pixels=True)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/z3h53fn0184inh06pynjilh4q2psklx3-python3.11-torchrl-0.2.1/lib/python3.11/site-packages/torchrl/envs/libs/gym.py", line 399, in __call__
    instance: GymWrapper = super().__call__(*args, **kwargs)
                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/z3h53fn0184inh06pynjilh4q2psklx3-python3.11-torchrl-0.2.1/lib/python3.11/site-packages/torchrl/envs/common.py", line 137, in __call__
    instance: EnvBase = super().__call__(*args, **kwargs)
                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/z3h53fn0184inh06pynjilh4q2psklx3-python3.11-torchrl-0.2.1/lib/python3.11/site-packages/torchrl/envs/libs/gym.py", line 957, in __init__
    super().__init__(**kwargs)
  File "/nix/store/z3h53fn0184inh06pynjilh4q2psklx3-python3.11-torchrl-0.2.1/lib/python3.11/site-packages/torchrl/envs/libs/gym.py", line 508, in __init__
    super().__init__(**kwargs)
  File "/nix/store/z3h53fn0184inh06pynjilh4q2psklx3-python3.11-torchrl-0.2.1/lib/python3.11/site-packages/torchrl/envs/common.py", line 2171, in __init__
    self._env = self._build_env(**kwargs)  # writes the self._env attribute
                ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/z3h53fn0184inh06pynjilh4q2psklx3-python3.11-torchrl-0.2.1/lib/python3.11/site-packages/torchrl/envs/libs/gym.py", line 1025, in _build_env
    env = super()._build_env(env, pixels_only=pixels_only, from_pixels=from_pixels)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/z3h53fn0184inh06pynjilh4q2psklx3-python3.11-torchrl-0.2.1/lib/python3.11/site-packages/torchrl/envs/libs/gym.py", line 586, in _build_env
    env = self._build_gym_env(env, pixels_only)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/z3h53fn0184inh06pynjilh4q2psklx3-python3.11-torchrl-0.2.1/lib/python3.11/site-packages/torchrl/envs/libs/gym.py", line 616, in _build_gym_env
    return pixel_observation.PixelObservationWrapper(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/42khxdyfdn74a851v3znz0caqjlgy3ih-python3.11-gym-0.26.2/lib/python3.11/site-packages/gym/wrappers/pixel_observation.py", line 138, in __init__
    pixels = self._render(**render_kwargs[pixel_key])
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/42khxdyfdn74a851v3znz0caqjlgy3ih-python3.11-gym-0.26.2/lib/python3.11/site-packages/gym/wrappers/pixel_observation.py", line 204, in _render
    render = self.env.render(*args, **kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/42khxdyfdn74a851v3znz0caqjlgy3ih-python3.11-gym-0.26.2/lib/python3.11/site-packages/gym/core.py", line 329, in render
    return self.env.render(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/42khxdyfdn74a851v3znz0caqjlgy3ih-python3.11-gym-0.26.2/lib/python3.11/site-packages/gym/wrappers/order_enforcing.py", line 51, in render
    return self.env.render(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/42khxdyfdn74a851v3znz0caqjlgy3ih-python3.11-gym-0.26.2/lib/python3.11/site-packages/gym/envs/mujoco/mujoco_env.py", line 391, in render
    self._get_viewer(self.render_mode).render(camera_id=camera_id)
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/42khxdyfdn74a851v3znz0caqjlgy3ih-python3.11-gym-0.26.2/lib/python3.11/site-packages/gym/envs/mujoco/mujoco_env.py", line 426, in _get_viewer
    self.viewer = RenderContextOffscreen(self.model, self.data)
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/42khxdyfdn74a851v3znz0caqjlgy3ih-python3.11-gym-0.26.2/lib/python3.11/site-packages/gym/envs/mujoco/mujoco_rendering.py", line 232, in __init__
    super().__init__(model, data, offscreen=True)
  File "/nix/store/42khxdyfdn74a851v3znz0caqjlgy3ih-python3.11-gym-0.26.2/lib/python3.11/site-packages/gym/envs/mujoco/mujoco_rendering.py", line 57, in __init__
    self.con = mujoco.MjrContext(self.model, mujoco.mjtFontScale.mjFONTSCALE_150)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
mujoco.FatalError: an OpenGL platform library has not been loaded into this process, this most likely means that a valid OpenGL context has not been created before mjr_makeContext was called
----------------------------- Captured stdout call -----------------------------
check_env_specs succeeded!
_____________________ TestTensorboard.test_log_video[None] _____________________
Traceback (most recent call last):
  File "/nix/store/xbsrnrzlv658h5r85pkw6v17ns63wlnx-python3.11-pytest-7.4.3/lib/python3.11/site-packages/_pytest/runner.py", line 341, in from_call
    result: Optional[TResult] = func()
                                ^^^^^^
  File "/nix/store/xbsrnrzlv658h5r85pkw6v17ns63wlnx-python3.11-pytest-7.4.3/lib/python3.11/site-packages/_pytest/runner.py", line 262, in <lambda>
    lambda: ihook(item=item, **kwds), when=when, reraise=reraise
            ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/zsnbysi65541mp54akjigfg9vw88xjvc-python3.11-pluggy-1.3.0/lib/python3.11/site-packages/pluggy/_hooks.py", line 493, in __call__
    return self._hookexec(self.name, self._hookimpls, kwargs, firstresult)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/zsnbysi65541mp54akjigfg9vw88xjvc-python3.11-pluggy-1.3.0/lib/python3.11/site-packages/pluggy/_manager.py", line 115, in _hookexec
    return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/zsnbysi65541mp54akjigfg9vw88xjvc-python3.11-pluggy-1.3.0/lib/python3.11/site-packages/pluggy/_callers.py", line 152, in _multicall
    return outcome.get_result()
           ^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/zsnbysi65541mp54akjigfg9vw88xjvc-python3.11-pluggy-1.3.0/lib/python3.11/site-packages/pluggy/_result.py", line 114, in get_result
    raise exc.with_traceback(exc.__traceback__)
  File "/nix/store/zsnbysi65541mp54akjigfg9vw88xjvc-python3.11-pluggy-1.3.0/lib/python3.11/site-packages/pluggy/_callers.py", line 77, in _multicall
    res = hook_impl.function(*args)
          ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/xbsrnrzlv658h5r85pkw6v17ns63wlnx-python3.11-pytest-7.4.3/lib/python3.11/site-packages/_pytest/runner.py", line 177, in pytest_runtest_call
    raise e
  File "/nix/store/xbsrnrzlv658h5r85pkw6v17ns63wlnx-python3.11-pytest-7.4.3/lib/python3.11/site-packages/_pytest/runner.py", line 169, in pytest_runtest_call
    item.runtest()
  File "/nix/store/xbsrnrzlv658h5r85pkw6v17ns63wlnx-python3.11-pytest-7.4.3/lib/python3.11/site-packages/_pytest/python.py", line 1792, in runtest
    self.ihook.pytest_pyfunc_call(pyfuncitem=self)
  File "/nix/store/zsnbysi65541mp54akjigfg9vw88xjvc-python3.11-pluggy-1.3.0/lib/python3.11/site-packages/pluggy/_hooks.py", line 493, in __call__
    return self._hookexec(self.name, self._hookimpls, kwargs, firstresult)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/zsnbysi65541mp54akjigfg9vw88xjvc-python3.11-pluggy-1.3.0/lib/python3.11/site-packages/pluggy/_manager.py", line 115, in _hookexec
    return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/zsnbysi65541mp54akjigfg9vw88xjvc-python3.11-pluggy-1.3.0/lib/python3.11/site-packages/pluggy/_callers.py", line 113, in _multicall
    raise exception.with_traceback(exception.__traceback__)
  File "/nix/store/zsnbysi65541mp54akjigfg9vw88xjvc-python3.11-pluggy-1.3.0/lib/python3.11/site-packages/pluggy/_callers.py", line 77, in _multicall
    res = hook_impl.function(*args)
          ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/xbsrnrzlv658h5r85pkw6v17ns63wlnx-python3.11-pytest-7.4.3/lib/python3.11/site-packages/_pytest/python.py", line 194, in pytest_pyfunc_call
    result = testfunction(**testargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/build/source/test/test_loggers.py", line 90, in test_log_video
    tb_logger.log_video(
  File "/nix/store/z3h53fn0184inh06pynjilh4q2psklx3-python3.11-torchrl-0.2.1/lib/python3.11/site-packages/torchrl/record/loggers/tensorboard.py", line 87, in log_video
    self.experiment.add_video(
  File "/nix/store/kf5sg15b6w2gw2dxz9q8x5fv62qramfv-python3.11-torch-2.1.2/lib/python3.11/site-packages/torch/utils/tensorboard/writer.py", line 805, in add_video
    video(tag, vid_tensor, fps), global_step, walltime
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/kf5sg15b6w2gw2dxz9q8x5fv62qramfv-python3.11-torch-2.1.2/lib/python3.11/site-packages/torch/utils/tensorboard/summary.py", line 624, in video
    video = make_video(tensor, fps)
            ^^^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/kf5sg15b6w2gw2dxz9q8x5fv62qramfv-python3.11-torch-2.1.2/lib/python3.11/site-packages/torch/utils/tensorboard/summary.py", line 651, in make_video
    clip.write_gif(filename, verbose=False, logger=None)
  File "/nix/store/rq0bzxalhbamysnx700x9mwmzr0cyid8-python3.11-decorator-5.1.1/lib/python3.11/site-packages/decorator.py", line 232, in fun
    return caller(func, *(extras + args), **kw)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/xh6kx7z6g2z7yyxy5f9mgl1znxc3v35z-python3.11-moviepy-1.0.3/lib/python3.11/site-packages/moviepy/decorators.py", line 54, in requires_duration
    return f(clip, *a, **k)
           ^^^^^^^^^^^^^^^^
  File "/nix/store/rq0bzxalhbamysnx700x9mwmzr0cyid8-python3.11-decorator-5.1.1/lib/python3.11/site-packages/decorator.py", line 232, in fun
    return caller(func, *(extras + args), **kw)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/xh6kx7z6g2z7yyxy5f9mgl1znxc3v35z-python3.11-moviepy-1.0.3/lib/python3.11/site-packages/moviepy/decorators.py", line 22, in convert_masks_to_RGB
    return f(clip, *a, **k)
           ^^^^^^^^^^^^^^^^
  File "/nix/store/xh6kx7z6g2z7yyxy5f9mgl1znxc3v35z-python3.11-moviepy-1.0.3/lib/python3.11/site-packages/moviepy/video/VideoClip.py", line 432, in write_gif
    write_gif_with_image_io(self, filename, fps=fps, opt=opt, loop=loop,
  File "/nix/store/xh6kx7z6g2z7yyxy5f9mgl1znxc3v35z-python3.11-moviepy-1.0.3/lib/python3.11/site-packages/moviepy/video/io/gif_writers.py", line 285, in write_gif_with_image_io
    writer.append_data(frame)
  File "/nix/store/qja4xicfhf7z318anr6j6md9gx2nq1xs-python3.11-imageio-2.33.1/lib/python3.11/site-packages/imageio/v2.py", line 226, in append_data
    return self.instance.write(im, **self.write_args)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/qja4xicfhf7z318anr6j6md9gx2nq1xs-python3.11-imageio-2.33.1/lib/python3.11/site-packages/imageio/plugins/pillow.py", line 433, in write
    raise ValueError("Can't write images with one color channel.")
ValueError: Can't write images with one color channel.
------------------------------ Captured log call -------------------------------
WARNING  py.warnings:warnings.py:109 /nix/store/xh6kx7z6g2z7yyxy5f9mgl1znxc3v35z-python3.11-moviepy-1.0.3/lib/python3.11/site-packages/moviepy/video/fx/painting.py:7: DeprecationWarning: Please use `sobel` from the `scipy.ndimage` namespace, the `scipy.ndimage.filters` namespace is deprecated.
  from scipy.ndimage.filters import sobel

WARNING  py.warnings:warnings.py:109 <frozen importlib._bootstrap>:241: RuntimeWarning: Your system is avx2 capable but pygame was not built with support for it. The performance of some of your blits could be adversely affected. Consider enabling compile time detection with environment variables like PYGAME_DETECT_AVX2=1 if you are compiling without cross compilation.
____________________ TestTensorboard.test_log_video[steps1] ____________________
Traceback (most recent call last):
  File "/nix/store/xbsrnrzlv658h5r85pkw6v17ns63wlnx-python3.11-pytest-7.4.3/lib/python3.11/site-packages/_pytest/runner.py", line 341, in from_call
    result: Optional[TResult] = func()
                                ^^^^^^
  File "/nix/store/xbsrnrzlv658h5r85pkw6v17ns63wlnx-python3.11-pytest-7.4.3/lib/python3.11/site-packages/_pytest/runner.py", line 262, in <lambda>
    lambda: ihook(item=item, **kwds), when=when, reraise=reraise
            ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/zsnbysi65541mp54akjigfg9vw88xjvc-python3.11-pluggy-1.3.0/lib/python3.11/site-packages/pluggy/_hooks.py", line 493, in __call__
    return self._hookexec(self.name, self._hookimpls, kwargs, firstresult)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/zsnbysi65541mp54akjigfg9vw88xjvc-python3.11-pluggy-1.3.0/lib/python3.11/site-packages/pluggy/_manager.py", line 115, in _hookexec
    return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/zsnbysi65541mp54akjigfg9vw88xjvc-python3.11-pluggy-1.3.0/lib/python3.11/site-packages/pluggy/_callers.py", line 152, in _multicall
    return outcome.get_result()
           ^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/zsnbysi65541mp54akjigfg9vw88xjvc-python3.11-pluggy-1.3.0/lib/python3.11/site-packages/pluggy/_result.py", line 114, in get_result
    raise exc.with_traceback(exc.__traceback__)
  File "/nix/store/zsnbysi65541mp54akjigfg9vw88xjvc-python3.11-pluggy-1.3.0/lib/python3.11/site-packages/pluggy/_callers.py", line 77, in _multicall
    res = hook_impl.function(*args)
          ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/xbsrnrzlv658h5r85pkw6v17ns63wlnx-python3.11-pytest-7.4.3/lib/python3.11/site-packages/_pytest/runner.py", line 177, in pytest_runtest_call
    raise e
  File "/nix/store/xbsrnrzlv658h5r85pkw6v17ns63wlnx-python3.11-pytest-7.4.3/lib/python3.11/site-packages/_pytest/runner.py", line 169, in pytest_runtest_call
    item.runtest()
  File "/nix/store/xbsrnrzlv658h5r85pkw6v17ns63wlnx-python3.11-pytest-7.4.3/lib/python3.11/site-packages/_pytest/python.py", line 1792, in runtest
    self.ihook.pytest_pyfunc_call(pyfuncitem=self)
  File "/nix/store/zsnbysi65541mp54akjigfg9vw88xjvc-python3.11-pluggy-1.3.0/lib/python3.11/site-packages/pluggy/_hooks.py", line 493, in __call__
    return self._hookexec(self.name, self._hookimpls, kwargs, firstresult)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/zsnbysi65541mp54akjigfg9vw88xjvc-python3.11-pluggy-1.3.0/lib/python3.11/site-packages/pluggy/_manager.py", line 115, in _hookexec
    return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/zsnbysi65541mp54akjigfg9vw88xjvc-python3.11-pluggy-1.3.0/lib/python3.11/site-packages/pluggy/_callers.py", line 113, in _multicall
    raise exception.with_traceback(exception.__traceback__)
  File "/nix/store/zsnbysi65541mp54akjigfg9vw88xjvc-python3.11-pluggy-1.3.0/lib/python3.11/site-packages/pluggy/_callers.py", line 77, in _multicall
    res = hook_impl.function(*args)
          ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/xbsrnrzlv658h5r85pkw6v17ns63wlnx-python3.11-pytest-7.4.3/lib/python3.11/site-packages/_pytest/python.py", line 194, in pytest_pyfunc_call
    result = testfunction(**testargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/build/source/test/test_loggers.py", line 90, in test_log_video
    tb_logger.log_video(
  File "/nix/store/z3h53fn0184inh06pynjilh4q2psklx3-python3.11-torchrl-0.2.1/lib/python3.11/site-packages/torchrl/record/loggers/tensorboard.py", line 87, in log_video
    self.experiment.add_video(
  File "/nix/store/kf5sg15b6w2gw2dxz9q8x5fv62qramfv-python3.11-torch-2.1.2/lib/python3.11/site-packages/torch/utils/tensorboard/writer.py", line 805, in add_video
    video(tag, vid_tensor, fps), global_step, walltime
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/kf5sg15b6w2gw2dxz9q8x5fv62qramfv-python3.11-torch-2.1.2/lib/python3.11/site-packages/torch/utils/tensorboard/summary.py", line 624, in video
    video = make_video(tensor, fps)
            ^^^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/kf5sg15b6w2gw2dxz9q8x5fv62qramfv-python3.11-torch-2.1.2/lib/python3.11/site-packages/torch/utils/tensorboard/summary.py", line 651, in make_video
    clip.write_gif(filename, verbose=False, logger=None)
  File "/nix/store/rq0bzxalhbamysnx700x9mwmzr0cyid8-python3.11-decorator-5.1.1/lib/python3.11/site-packages/decorator.py", line 232, in fun
    return caller(func, *(extras + args), **kw)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/xh6kx7z6g2z7yyxy5f9mgl1znxc3v35z-python3.11-moviepy-1.0.3/lib/python3.11/site-packages/moviepy/decorators.py", line 54, in requires_duration
    return f(clip, *a, **k)
           ^^^^^^^^^^^^^^^^
  File "/nix/store/rq0bzxalhbamysnx700x9mwmzr0cyid8-python3.11-decorator-5.1.1/lib/python3.11/site-packages/decorator.py", line 232, in fun
    return caller(func, *(extras + args), **kw)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/xh6kx7z6g2z7yyxy5f9mgl1znxc3v35z-python3.11-moviepy-1.0.3/lib/python3.11/site-packages/moviepy/decorators.py", line 22, in convert_masks_to_RGB
    return f(clip, *a, **k)
           ^^^^^^^^^^^^^^^^
  File "/nix/store/xh6kx7z6g2z7yyxy5f9mgl1znxc3v35z-python3.11-moviepy-1.0.3/lib/python3.11/site-packages/moviepy/video/VideoClip.py", line 432, in write_gif
    write_gif_with_image_io(self, filename, fps=fps, opt=opt, loop=loop,
  File "/nix/store/xh6kx7z6g2z7yyxy5f9mgl1znxc3v35z-python3.11-moviepy-1.0.3/lib/python3.11/site-packages/moviepy/video/io/gif_writers.py", line 285, in write_gif_with_image_io
    writer.append_data(frame)
  File "/nix/store/qja4xicfhf7z318anr6j6md9gx2nq1xs-python3.11-imageio-2.33.1/lib/python3.11/site-packages/imageio/v2.py", line 226, in append_data
    return self.instance.write(im, **self.write_args)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/qja4xicfhf7z318anr6j6md9gx2nq1xs-python3.11-imageio-2.33.1/lib/python3.11/site-packages/imageio/plugins/pillow.py", line 433, in write
    raise ValueError("Can't write images with one color channel.")
ValueError: Can't write images with one color channel.
```

</details>

Reply:
I guess it appears only here because it's a versioning issue? 
But making tensorboard logger robust to 1d image seems to be the right thing to do, I can make a patch for that too.

Reply:
We are using `tensorboard==2.15.1` though.

Reply:
> But making tensorboard logger robust to 1d image seems to be the right thing to do, I can make a patch for that too.

Is [this](https://github.com/pytorch/rl/commit/24d14ad4ec405f1c827284e360a02c02006663eb) the patch in question ?
Also, when do you think that the next release of torchrl will be out ?

Reply:
```nix
    # TypeError: _FlakyPlugin._make_test_flaky() got an unexpected keyword argument 'reruns'
    "test_parallel_env_transform_consistency"
```
This error was due to the inclusion of `flaky` instead of `pytest-rerunfailures`.

Reply:
happy to swap flaky for rerunfailures if that helps

next release in 1w

Do not hesitate to reach out privately to let me know how I can help with this!

Reply:
> happy to swap flaky for rerunfailures if that helps

The torchrl tests actually expect the `pytest-rerunfailures` plugin which provides `@pytest.mark.flaky`.
At first, I thought that the dependency was `flaky` (which also exposes a `@pytest.mark.flaky`, but a different one). This was wrong and causing the test to error.
Including the correct dependency made it work.



> next release in 1w

Cool ! We will then be able to remove the `fetchpatch` statements. For now, it will do ;)

> Do not hesitate to reach out privately to let me know how I can help with this!

Thank you so much for your help !